platform/darwin/src/MGLLoggingConfiguration.h

```diff
@@ -0,0 +1,83 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLFoundation.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Constants indicating the message's logging level.
+ */
+typedef NS_ENUM(NSInteger, MGLLoggingLevel) {
+    /**
+     None-level messages are ignored.
+     */
+    MGLLoggingLevelNone = 0,
+    /**
+     Info-level messages contain information that may be helpful for flow tracing
+     but is not essential.
+     */
+    MGLLoggingLevelInfo,
+    /**
+     Debug-level messages contain information that may be helpul for troubleshooting
+     specific problems.
+     */
+    MGLLoggingLevelDebug,
+    /**
+     Error-level messages contain information that is intended to aid in process-level
+     errors.
+     */
+    MGLLoggingLevelError,
+    /**
+     Fault-level messages contain system-level error information.
+     */
+    MGLLoggingLevelFault,
+    /**
+     :nodoc: Any new logging level should be included in the default logging implementation.
+     */
+};
+
+/**
+ A block to be called once `loggingLevel` is set to a higher value than `MGLLoggingLevelNone`.
+ 
+ @param loggingLevel The message logging level.
+ @param filePath The description of the file and method for the calling message.
+ @param message The logging message.
+ */
+typedef void (^MGLLoggingBlockHandler)(MGLLoggingLevel loggingLevel, NSString *filePath, NSUInteger line, NSString *message);
+
+/**
+ The `MGLLoggingConfiguration` object provides a global way to set this SDK logging levels
+ and logging handler.
+ */
+MGL_EXPORT
+@interface MGLLoggingConfiguration : NSObject
+
+/**
+ The handler this SDK uses to log messages.
+ 
+ If this property is set to nil or if no custom handler is provided this property
+ is set to the default handler.
+ 
+ The default handler uses `os_log` and `NSLog` for iOS 10+ and iOS < 10 respectively.
```

Comment:
I think it's worth mentioning that arguments passed in (and passed to os_log) will have the `{public}` specifier.

Reply:
I changed that and we are not using `{public}` anymore:
`os_log_with_type(mapbox_log, logType, "%@ - %d: %@", fileName, line, message);`